test/tests/arch.bats

```diff
@@ -36,3 +36,23 @@ load ../util/util
     [[ "${status}" == "12" ]]
     [[ "${output}" == "[!] testpkg is not available for the 'amd64' architecture." ]]
 }
+@test "allow arch all in sub packages" {
+    package_testpkg-doc() {
+      arch=('all')
+    }
+
+    package_testpkg() {
+      echo ''
+    }
+
+    pkgbuild string maintainer1 'Foo Bar <foo@bar.com>'
+    pkgbuild array pkgname testpkg testpkg-doc
+    pkgbuild string pkgver 1.0.0
+    pkgbuild string pkgrel 1
+    pkgbuild string pkgdesc "package description"
+    pkgbuild array arch amd64
+    pkgbuild function package_testpkg
+    pkgbuild function package_testpkg-doc
+    pkgbuild clean
+    makedeb --lint
```

Comment:
You should probably just run `makedeb`, and then check that the `Architecture` field in the control file for `package_testpkg-doc` is set correctly.

This contains an example of how you might check the built package for such info: https://github.com/makedeb/makedeb/blob/21095c90a5935489ec70f6ac3b69e971b13cf32f/test/tests/depends.bats#L53

Reply:
I'd also like a test added that makes sure any changes made to `arch` aren't made permanent outside of the specific `package` function - they shouldn't, but just making sure they aren't with a test would be nice.